ggml-rpc.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "ggml.h"
+#include "ggml-backend.h"
+#include <string>
+#include <memory>
+#ifdef _WIN32
+#  define WIN32_LEAN_AND_MEAN
+#  ifndef NOMINMAX
+#     define NOMINMAX
+#  endif
+#  include <windows.h>
+#  include <winsock2.h>
+#endif
```

Comment:
Not strictly necessary to fix this now, but it's not great to pull all of these headers, or to expose the internal structs of the RPC backend in this header. The `socket_t` class will prevent using the RPC backend in purely C programs. Maybe the API could be streamlined, or these details moved to a private header.

Reply:
Yes, putting implementation details in the `ggml-rpc.h` was a mistake. I have removed all internal structs and networking definitions and introduced `GGML_API GGML_CALL void start_rpc_server(ggml_backend_t backend, const char * endpoint, size_t free_mem, size_t total_mem);`